### PR TITLE
docs: add content for Gas Tank API Documentation (XDEFI-6907)

### PR DESCRIPTION
**Pull request by HoangVD2**
Ticket: https://xdefi-tech.atlassian.net/browse/XDEFI-6907

### CODE REVIEW

File: gas-tank/balance-services.md [hunk 1]
```diff
@@ -0,0 +1,515 @@
...
+  });
+```
+
+:::
+
+### Consume balance
+
+This endpoint allows users to consume their balance on the Gas Tank platform.
+
+To consume the balance, you need to sign the message with the private key of the address you want to consume the balance from. The `signature` field is the signature of the message signed by the address owner.
```
Comment (davidp94) [inline gas-tank/balance-services.md:236]: Can you expand (for all the consume, deposit, withdraw) with
- What message (how to get it via which endpoint) and provide snippet
- How to sign it using ethers.js for instance (check the demo app)

**Review by davidp94 — CHANGES_REQUESTED** [review]

File: gas-tank/balance-services.md (hunk 1 above)
Comment (hvu8a5) [reply gas-tank/balance-services.md:253]: added for all @davidp94

Comment (HoangVD2) [reply gas-tank/balance-services.md:253]: added for all @davidp94: https://github.com/XDeFi-tech/docs-new/pull/34/commits/b18f65e19c4eaa2a6b456b248600dd9f5a7fd901

File: gas-tank/authentication-services.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
+## Authentication
+
+### Multiple Address Login
+
+This endpoint facilitates the generation of JWT tokens for multiple wallet addresses in a single request. The request payload should consist of an array of objects, each containing an address and its corresponding signature. Upon successful validation of the signatures, the server will generate JWT tokens for the provided addresses.
+
+::: code-group
```
Comment (davidp94) [inline gas-tank/authentication-services.md:7]: Can you specify what are the messages to sign here (how to get it? via API) and how to sign them?

File: gas-tank/balance-services.md [hunk 3]
```diff
@@ -0,0 +1,540 @@
...
+
+You can sign a string of data using [ethers.js](https://docs.ethers.org/v5), below is an example of [how to sign a message using ethers.js](https://docs.ethers.org/v5/getting-started/#getting-started--signing) and increase the balance.
+
+::: code-group
+
+```javascript [Request]
+import { ethers } from "ethers";
+
+const GAS_TANK_ENDPOINT = "https://gas-tank.xdefi.services";
+const message = "0x1234"; // Message to sign // [!code highlight]
```
Comment (davidp94) [inline gas-tank/balance-services.md:134]: Can you specify what is the messages to sign here (how to get it? via API)

File: gas-tank/balance-services.md [hunk 4]
```diff
@@ -0,0 +1,540 @@
...
+
+Same as the [Increase balance](#increase-balance), you can sign a string of data using [ethers.js](https://docs.ethers.org/v5/getting-started/), below is an example of [how to sign a message using ethers.js](https://docs.ethers.org/v5/getting-started/#getting-started--signing) and withdraw the balance.
+
+::: code-group
+
+```javascript [Request]
+import { ethers } from "ethers";
+
+const GAS_TANK_ENDPOINT = "https://gas-tank.xdefi.services";
+const message = "0x1234"; // Message to sign // [!code highlight]
```
Comment (davidp94) [inline gas-tank/balance-services.md:183]: Can you specify what is the messages to sign here (how to get it? via API)

File: gas-tank/balance-services.md [hunk 5]
```diff
@@ -0,0 +1,540 @@
...
+    "Content-Type": "application/json",
+    Authorization: `Bearer ${jwtToken}`, // JWT token // [!code highlight]
+  },
+  body: {
+    address: "string", // [!code highlight]
+    tokenAddress: "string", // [!code highlight]
+    amount: "string", // [!code highlight]
+    chain: "string", // [!code highlight]
+    recipient: "string", // [!code highlight]
+    message: "string", // [!code highlight]
```
Comment (davidp94) [inline gas-tank/balance-services.md:235]: Can you specify what is the messages to sign here (how to get it? via API) and how to sign it

File: gas-tank/balance-services.md [hunk 6]
```diff
@@ -0,0 +1,540 @@
...
+
+Same as the [Increase balance](#increase-balance)/[Withdraw balance](#withdraw-balance), you can sign a string of data using [ethers.js](https://docs.ethers.org/v5/getting-started/), below is an example of [how to sign a message using ethers.js](https://docs.ethers.org/v5/getting-started/#getting-started--signing) and consume the balance.
+
+::: code-group
+
+```javascript [Request]
+import { ethers } from "ethers";
+
+const GAS_TANK_ENDPOINT = "https://gas-tank.xdefi.services";
+const message = "0x1234"; // Message to sign // [!code highlight]
```
Comment (davidp94) [inline gas-tank/balance-services.md:263]: Can you specify what is the messages to sign here (how to get it? via API) and how to sign it

File: gas-tank/balance-services.md [hunk 7]
```diff
@@ -0,0 +1,540 @@
...
+    }
+  ]
+}
+```
+
+:::
+
+### Get user balance queue update
+
+This endpoint allows users to get the balance queue update for a specific address.
```
Comment (davidp94) [inline gas-tank/balance-services.md:341]: can you explain that it is to get the status of any job ongoing (deposit, consume, internal transfer, withdrawal)

File: gas-tank/balance-services.md [hunk 8]
```diff
@@ -0,0 +1,540 @@
...
+  ],
+  ...
+}
+```
+
+:::
+
+### Get balance queue update by ID
+
+This endpoint allows users to get the balance queue update for a specific ID.
```
Comment (davidp94) [inline gas-tank/balance-services.md:403]: can you explain that it is to get the status of any job ongoing (deposit, consume, internal transfer, withdrawal)

File: gas-tank/balance-services.md [hunk 9]
```diff
@@ -0,0 +1,540 @@
...
+  },
+  ...
+]
+```
+
+:::
+
+### Increase balance
+
+This endpoint allows users to increase their balance on the Gas Tank platform.
```
Comment (davidp94) [inline gas-tank/balance-services.md:122]: Can you say in other words it's for depositing tokens; can we add a table of the confirmation count for it ?

cc. @VachaganGrigoryan

**Review by davidp94 — CHANGES_REQUESTED** [review]

File: gas-tank/balance-services.md [hunk 10]
```diff
@@ -0,0 +1,554 @@
...
+
+::: code-group
+
+```javascript [Request]
+import { ethers } from "ethers";
+
+const web3 = new Web3(window.ethereum);
+const GAS_TANK_ENDPOINT = "https://gas-tank.xdefi.services";
+const address = "0x1234567890123456789012345678901234567890"; // Address to deposit balance // [!code highlight]
+const message = web3.utils.sha3("Deposit balance to Gas Tank"); // Message to sign // [!code highlight]
```
Comment (davidp94) [inline gas-tank/balance-services.md:136]: Should use this endpoint.

https://gas-tank.xdefi.services/docs/#/Messages/getDepositMessageController

File: gas-tank/balance-services.md [hunk 11]
```diff
@@ -0,0 +1,554 @@
...
+const privateKey = "0x1234"; // Private key of the address // [!code highlight]
+const wallet = new ethers.Wallet(privateKey);
+const signature = await wallet.signMessage(message);
```
Comment (davidp94) [inline gas-tank/balance-services.md:139]: Check full example: https://github.com/XDeFi-tech/gas_tank_service/blob/main/demo/src/components/Forms/DepositForm.tsx#L103-L136

File: gas-tank/balance-services.md [hunk 12]
```diff
@@ -0,0 +1,554 @@
...
+Same as the [Deposit balance](#deposit-balance), you can sign a string of data using [ethers.js](https://docs.ethers.org/v5/getting-started/), below is an example of [how to sign a message using ethers.js](https://docs.ethers.org/v5/getting-started/#getting-started--signing) and withdraw the balance.
+
+::: code-group
+
+```javascript [Request]
+import { ethers } from "ethers";
+
+const web3 = new Web3(window.ethereum);
+const GAS_TANK_ENDPOINT = "https://gas-tank.xdefi.services";
+const address = "0x1234567890123456789012345678901234567890"; // Address to withdraw balance // [!code highlight]
```
Comment (davidp94) [inline gas-tank/balance-services.md:186]: https://github.com/XDeFi-tech/gas_tank_service/blob/main/demo/src/components/Forms/WithdrawForm.tsx#L97-L115

File: gas-tank/balance-services.md [hunk 13]
```diff
@@ -0,0 +1,554 @@
...
+
+::: code-group
+
+```javascript [Request]
+import { ethers } from "ethers";
+
+const web3 = new Web3(window.ethereum);
+const GAS_TANK_ENDPOINT = "https://gas-tank.xdefi.services";
+const address = "0x1234567890123456789012345678901234567890"; // Address to transfer balance // [!code highlight]
+const message = web3.utils.sha3("Transfer balance from Gas Tank"); // Message to sign // [!code highlight]
```
Comment (davidp94) [inline gas-tank/balance-services.md:230]: https://github.com/XDeFi-tech/gas_tank_service/blob/main/demo/src/components/Forms/InternalTransferForm.tsx

File: gas-tank/balance-services.md [hunk 14]
```diff
@@ -0,0 +1,554 @@
...
+
+::: code-group
+
+```javascript [Request]
+import { ethers } from "ethers";
+
+const web3 = new Web3(window.ethereum);
+const GAS_TANK_ENDPOINT = "https://gas-tank.xdefi.services";
+const address = "0x1234567890123456789012345678901234567890"; // Address to consume balance // [!code highlight]
+const message = web3.utils.sha3("Consume balance from Gas Tank"); // Message to sign // [!code highlight]
```
Comment (davidp94) [inline gas-tank/balance-services.md:277]: https://github.com/XDeFi-tech/gas_tank_service/blob/main/demo/src/components/Forms/ConsumeForm.tsx

File: gas-tank/balance-services.md [hunk 15]
```diff
@@ -0,0 +1,554 @@
...
+  .catch((error) => {
+    // Catch & handle the error
+  });
+```
+
+:::
+
+### Get a quote for consuming balance
+
+This endpoint allows users to generate a quote for consuming their balance on the Gas Tank platform.
```
Comment (davidp94) [inline gas-tank/balance-services.md:309]: We should put the quote before the Consume endpoint

File: gas-tank/balance-services.md [hunk 16]
```diff
@@ -0,0 +1,554 @@
...
+
+::: code-group
+
+```javascript [Request]
+const GAS_TANK_ENDPOINT = "https://gas-tank.xdefi.services";
+
+await fetch(`${GAS_TANK_ENDPOINT}/balances/consume/quote`, {
```
Comment (davidp94) [inline gas-tank/balance-services.md:316]: https://github.com/XDeFi-tech/gas_tank_service/blob/main/demo/src/components/Forms/ConsumeForm.tsx#L137

**Review by davidp94 — CHANGES_REQUESTED** [review]

**Review by davidp94 — CHANGES_REQUESTED** [review]

**Comment by HoangVD2** [comment]
Could you approve? @davidp94

**Review by davidp94 — APPROVED** [review]

**Pull request merged**